File: backend/bot.py
#!/usr/bin/env python

import urllib2, json, random, sys, os

def prompt():
    recv(raw_input(">>> ").lower())

def post(msg):
    print(msg)

def meme(srch):
    reqUrl = "http://api.pixplorer.co.uk/image?amount=1&size=tb&word=meme"
    if srch:
        memesrch = ""
        while srch:
            memesrch += srch[0] + "+"
            srch = srch[1:]
        memesrch = memesrch[:-1]
        reqUrl += memesrch
    memeUrl = eval(urllib2.urlopen(reqUrl).read())['images'][0]['imageurl']
    return "<a href='" + memeUrl + "'><img src='" + memeUrl + "' style='height: 130px;' /></a>"


def gif(srch):
    if srch:
        gif = eval(urllib2.urlopen("http://api.giphy.com/v1/gifs/search?q=" + srch + "&api_key=dc6zaTOxFJmzC").read())['data'][0]
        gifImg = "https://media.giphy.com/media/" + gif['id'] + "/giphy.gif"
    else:
        gif = eval(urllib2.urlopen("http://api.giphy.com/v1/gifs/random?api_key=dc6zaTOxFJmzC").read())['data']
        gifImg = gif['image_original_url']

    gifUrl = gif['url']
    return "<a href='" + gifUrl + "'><img src='" + gifImg + "' style='height: 130px;' /></a>"

def joke():
    return eval(urllib2.urlopen("https://gist.githubusercontent.com/neelusb/559916bf7ff7091a20c818e86534f4cc/raw/917c8baba74f6e3d323e95f14d7b681a0dfae5c3/jokes.json").read())[random.randint(0,7)]["joke"]
def recv(m):
    sessid = ""
    m = m.lower()
    mstr = m
    m = m.split()
    if "joke" in m:
        post(joke())
    elif "jokes" in m:
        post("Sorry, I can only tell one joke at a time. Here it is:")
        post(joke())
    if "meme" in m:
        meme("rnd123RND321rnd123RND321rnd")
    if m[0] == "exit" and len(m) == 1:
        sys.exit(0)
    if "gif" in m:
        if "about" in m:
            while m[0] != "about":
                m = m[1:]
            m = m[1:]
            gifsrch = ""
            while m:
                gifsrch += m[0] + "+"
                m = m[1:]
            gifsrch = gifsrch[:-1]
            post(gif(gifsrch))
        elif "of" in m:
            while m[0] != "of":
                m = m[1:]
            m = m[1:]
            gifsrch = ""
            while m:
                gifsrch += m[0] + "+"
                m = m[1:]
            gifsrch = gifsrch[:-1]
            post(gif(gifsrch))
        else:
            post(gif(False))

    elif "meme" in m:
        if "about" in m:
            while m[0] != "about":
                m = m[1:]
            m = m[1:]
            memesrch = ""
            while m:
                memesrch += m[0] + "+"
                m = m[1:]
            memesrch = memesrch[:-1]
            post(meme(memesrch))
        elif "of" in m:
            while m[0] != "of":
                m = m[1:]
            m = m[1:]
            memesrch = ""
            while m:
                memesrch += m[0] + "+"
                m = m[1:]
            memesrch = memesrch[:-1]
            post(meme(memesrch))
        else:
            post(meme(False))

    elif "yes" in m and os.path.isfile(sessid + "-HOGSbot_search.txt"):
        search = open(sessid + "-HOGSbot_search.txt", "r").read().split()
        searchout = ""
        for word in search:
            searchout += word + "+"
        searchout = searchout[:-1]
        post("<a href='https://www.google.com/search?q=" + searchout + "' target='_blank'>Click Here to Search for it in a new tab!</a>")
        os.remove(sessid + "-HOGSbot_search.txt")

    elif "no" in m and os.path.isfile("HOGSbot_search.txt"):
        post("Okay.")
        os.remove(sessid + "-HOGSbot_search.txt")

    else:
        post("I do not know what you mean. Would you like to search the web for "+ mstr +"?")
        open(sessid + "-HOGSbot_search.txt", "w").write(mstr)

if __name__ == "__main__":
    while 1:
        # try: prompt()
        # except: print; sys.exit(0)
        prompt()

File: scripts/bot.js
var name = getCookie("hbot-name");

function sendmsg() {
  $('.msgbox').append("<b>" + name + ":</b> " + $('.inputmsg').val() + '<br>');
  recv($('.inputmsg').val());
  $('.inputmsg').val("")
}

function post(msg) {
  $(".msgbox").append("<b>HOGSbot:</b> " + msg + "<br />");
}

function meme(srch) {
  var reqUrl = "http://api.pixplorer.co.uk/image?amount=1&size=tb&word=meme";
  var memesrch = "";
  while (srch) {
    memesrch += "+" + srch[0];
    srch = srch.slice(1);
  }
  reqUrl += memesrch;
  var memeUrl = "";

  return $.get(reqUrl, function( result ) {
    memeUrl = result['images'][0]['imageurl'];
    post("<a href='" + memeUrl + "' target='_blank'><img src='" + memeUrl + "' style='height: 130px;' /></a>");
  });
}

function joke() {
  $.get("https://gist.githubusercontent.com/neelusb/559916bf7ff7091a20c818e86534f4cc/raw/917c8baba74f6e3d323e95f14d7b681a0dfae5c3/jokes.json", function( result ) {
    post("<p>" + eval(result)[Math.floor((Math.random() * 9) + 1)]['joke'] + "</p>");
  });
}

function gif(srch) {
    if (srch) {
      $.get("http://api.giphy.com/v1/gifs/search?api_key=dc6zaTOxFJmzC&q=" + srch, function( result ) {
        gifv = result['data'][0];
        gifImg = "https://media.giphy.com/media/" + gifv['id'] + "/giphy.gif";
        gifUrl = gifv['url'];
        post("<a href='" + gifUrl + "' target='_blank'><img src='" + gifImg + "' style='height: 130px;' /></a>");
      });
    }
    else {
      $.get("http://api.giphy.com/v1/gifs/random?api_key=dc6zaTOxFJmzC", function( result ) {
        gifv = result['data'];
        gifImg = gifv['image_original_url'];
        gifUrl = gifv['url'];
        post("<a href='" + gifUrl + "' target='_blank'><img src='" + gifImg + "' style='height: 130px;' /></a>");
      });
    }
}

function inA(ele, arr) {
  if (arr.indexOf(ele) > -1) {
    return true;
  }
  else {
    return false;
  }
}

var hbsrch = "";

function recv(m) {
  var mstr = m;
  m = m.toLowerCase();
  m = m.split(" ");
  if (inA("joke", m)) {
    joke();
  }
  else if (inA("jokes", m)) {
    post("Sorry, I can only tell one joke at a time. Here it is:");
    joke();
  }

  else if (inA("gif", m)) {
    if (inA("about", m)) {
      while (m[0] != "about") {
        m = m.slice(1);
      }
      m = m.slice(1);
      var gifsrch = "";
      while (m.length > 0) {
        gifsrch += m[0] + "+";
        m = m.slice(1);
      }
      gifsrch = gifsrch.slice(0, -1);
      gif(gifsrch);
    }
    else if (inA("of", m)) {
      while (m[0] != "of") {
        m = m.slice(1);
      }
      m = m.slice(1);
      var gifsrch = "";
      while (m.length > 0) {
        gifsrch += m[0] + "+";
        m = m.slice(1);
      }
      gifsrch = gifsrch.slice(0, -1);
      gif(gifsrch);
    }
    else {
      gif();
    }
  }
  else if (inA("meme", m)) {
    if (inA("about", m)) {
      while (m[0] != "about") {
        m = m.slice(1);
      }
      m = m.slice(1);
      var memesrch = "";
      while (m.length > 0) {
        memesrch += m[0] + "+";
        m = m.slice(1);
      }
      memesrch = memesrch.slice(0, -1);
      meme(memesrch);
    }
    else if (inA("of", m)) {
      while (m[0] != "of") {
        m = m.slice(1);
      }
      m = m.slice(1);
      var memesrch = "";
      while (m.length > 0) {
        memesrch += m[0] + "+";
        m = m.slice(1);
      }
      memesrch = memesrch.slice(0, -1);
      meme(memesrch);
    }
    else {
      meme();
    }
  }
  else if (inA("yes", m) && hbsrch) {
    var search = hbsrch.split(" ");
    var searchout = "";
    for (i = 0; i < search.length; i++) {
      searchout += search[i] + "%20";
    }
    searchout = searchout.slice(0, -3);
    var searchoutinp = searchout;
    searchout = "";
    for (i = 0; i < searchoutinp.length; i++) {
      if (searchoutinp[i] == "+") {
        searchout += "%2B";
      }
      else {
        searchout += searchoutinp[i];
      }
    }
    post("Okay.");
    window.open('https://www.google.com/search?q=' + searchout, '_blank');
    hbsrch = false;
  }
  else if (inA("no", m) && hbsrch) {
    post("Okay.");
    hbsrch = false;
  }
  else {
    post("I do not know what you mean by " + mstr + ". Would you like to search the web for "+ mstr +"?");
    hbsrch = mstr;
  }
}
